dev/com.ibm.ws.security.spnego.fat.common/fat/src/com/ibm/ws/security/spnego/fat/config/ApacheKDCCommonTest.java

```diff
@@ -0,0 +1,1099 @@
+    }
+
+    /**
+     * Performs a call to the SPNEGO servlet that is expected to be successful using the common valid SPNEGO token. The
+     * response received is then verified and checked for the presence of GSS credentials.
+     *
+     * @return
+     * @throws Exception
+     */
+    public String commonSuccessfulSpnegoServletCall() throws Exception {
+        return successfulSpnegoServletCall(createCommonHeaders());
+    }
+
+    /**
+     * Performs a call to the SPNEGO servlet from an SSL client that is expected to be successful using the common
+     * valid SPNEGO token. The response received is then verified and checked for the presence of GSS credentials.
+     *
+     * @return
+     * @throws Exception
+     */
+    public String commonSuccessfulSpnegoServletCallSSLClient() throws Exception {
+        return successfulSpnegoServletCallSSLClient(createCommonHeaders());
+    }
+
+    /**
+     * Performs a call to the SPNEGO servlet that is expected to be unsuccessful using the common valid SPNEGO token,
+     * resulting in a 401. The response received is expected to be null due to the unauthorized request.
+     *
+     * @return
+     * @throws Exception
+     */
+    public String commonUnsuccessfulSpnegoServletCall() throws Exception {
+        return commonUnsuccessfulSpnegoServletCall(401);
+    }
+
+    /**
+     * Performs a call to the SPNEGO servlet that is expected to be unsuccessful using the common valid SPNEGO token.
+     * The response received is expected to be null due to the unauthorized request.
+     *
+     * @param expectedStatusCode
+     * @return
+     * @throws Exception
+     */
+    public String commonUnsuccessfulSpnegoServletCall(int expectedStatusCode) throws Exception {
+        return unsuccessfulSpnegoServletCall(createCommonHeaders(), SPNEGOConstants.IGNORE_ERROR_CONTENT, expectedStatusCode);
+    }
+
+    /**
+     * Creates and returns a map of headers using the SPNEGO token created for this test class in the Authorization
+     * header, Firefox as the User-Agent header value, TARGET_SERVER as the Host header value, and null as the remote
+     * address header value.
+     *
+     * @return
+     */
+    protected Map<String, String> createHeaders() {
+        return testHelper.setTestHeaders("Negotiate " + spnegoTokenForTestClass, SPNEGOConstants.FIREFOX, TARGET_SERVER, null);
+    }
+
+    /**
+     * Creates and returns a map of headers using the common SPNEGO token in the Authorization header, Firefox as the
+     * User-Agent header value, TARGET_SERVER as the Host header value, and null as the remote address header value.
+     *
+     * @return
+     * @throws Exception
+     */
+    protected Map<String, String> createCommonHeaders() throws Exception {
+        return testHelper.setTestHeaders("Negotiate " + getCommonSPNEGOToken(), SPNEGOConstants.FIREFOX, TARGET_SERVER, null);
+    }
+
+    protected static String getCommonSPNEGOToken() throws Exception {
```

Comment:
javadoc

Reply:
I'm not going to mark anymore Javadoc issues. Go through your code and add Javadoc to all the new methods and classes.